homeassistant/components/tesla/device_tracker.py

```diff
@@ -4,19 +4,35 @@
 from homeassistant.helpers.event import track_utc_time_change
 from homeassistant.util import slugify
 
+from homeassistant.components.device_tracker import see as dev_see
+
 from . import DOMAIN as TESLA_DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
 
 def setup_scanner(hass, config, see, discovery_info=None):
     """Set up the Tesla tracker."""
-    TeslaDeviceTracker(
+    hass.data[TESLA_DOMAIN]["devices"]["device_tracker"] = TeslaDeviceTracker(
         hass, config, see, hass.data[TESLA_DOMAIN]["devices"]["devices_tracker"]
     )
     return True
 
 
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up the Tesla binary_sensors by config_entry."""
+    return await hass.async_add_executor_job(
+        setup_scanner, hass, config_entry.data, dev_see, None
+    )
+
+
+async def async_unload_entry(hass, entry) -> bool:
+    """Unload a config entry."""
+    _LOGGER.debug("Unloading %s", hass.data[TESLA_DOMAIN]["devices"]["device_tracker"])
+    hass.data[TESLA_DOMAIN]["devices"]["device_tracker"].unload()
+    return True
+
+
 class TeslaDeviceTracker:
```

Comment:
Rewrite this to inherit from `TrackerEntity` imported from `device_tracker/config_entry.py`.

Reply:
Unfortunately the device tracker needs to change in the same PR as the config entry is activated since the new device tracker class is only supported by config entry based integrations.